compiler/vm/compilerbridge.nim

```diff
@@ -61,11 +63,22 @@ from compiler/ast/reports_internal import InternalReport
 from compiler/ast/report_enums import ReportKind
 
 type
+  ExecErrorKind* = enum
+    execErrorVm
+    execErrorVmGen
+    execErrorQuit
+
   ExecErrorReport* = object
-    stackTrace*: VmStackTrace ## The VM stack-trace
-    location*: TLineInfo      ## Source location of the trace
+    stackTrace*: VmStackTrace   ## The VM stack-trace
+    location*: TLineInfo        ## Source location of the trace
     instLoc*: InstantiationInfo ## report instantiation location
-    report*: VMReport         ## The report detailing the error
+    case kind*: ExecErrorKind   ## kind of error execution of vm code gen
+      of execErrorVm:
+        vmErr*: VmEvent
+      of execErrorVmGen:
+        genErr*: VmGenDiag
+      of execErrorQuit:
+        exitCode*: int
```

Comment:
@zerbina I evolved `ExecErrorReport` a little here, turns out I can drop the `VmEvent` this way, it's all coming off of yield reason anyhow. Thoughts?

Reply:
I believe you meant `VMReport`? Nonetheless, I like the change and think it's a good idea.

Reply:
Should have clarified, it was late last night.

When I said drop the `VmEvent`, I actually meant I no longer have the `VmEvtQuit` enum value and (`VmEventKind`) and corresponding `VmEvent` variant branch.

Reply:
Ah, I missed that part, thank you for clarifying. Yep, that's definitely better.